common/logisticspipes/network/NewGuiHandler.java

```diff
@@ -52,64 +38,28 @@ private NewGuiHandler() { }
 		return (T) NewGuiHandler.guimap.get(clazz).template();
 	}
 
-	@SuppressWarnings("unchecked")
-	@SneakyThrows({ IOException.class }) // , InvocationTargetException.class, IllegalAccessException.class, InstantiationException.class
-	// Suppression+sneakiness because these shouldn't ever fail, and if they do, it needs to fail.
-	public static final void initialize() {
-		final List<ClassInfo> classes = new ArrayList<>(ClassPath.from(NewGuiHandler.class.getClassLoader())
-				.getTopLevelClassesRecursive("logisticspipes.network.guis"));
+	public static void initialize() {
+		Set<Class<? extends GuiProvider>> classes = StaticResolverUtil.findClassesByType(GuiProvider.class);
 
-		loadGuiProvidersFromList(classes.stream().map(ClassInfo::getName).collect(Collectors.toList()));
+		loadGuiProviders(classes);
 
 		if (NewGuiHandler.guilist == null || NewGuiHandler.guilist.isEmpty()) {
-			System.err.println(
-					"LogisticsPipes could not search for its GuiProvider classes. Your mods are probably installed in a path containing spaces or special characters. Trying to use fallback solution.");
-			URL location = NewGuiHandler.class.getProtectionDomain().getCodeSource().getLocation();
-			String locationString = location.toString();
-			if (locationString.startsWith("jar:file:/") && locationString.contains("!")) {
-				locationString = locationString.substring(10, locationString.indexOf('!'));
-				locationString = java.net.URLDecoder.decode(locationString, "UTF-8");
-			}
-
-			List<String> classInfoList = new ArrayList<>();
-
-			File file = new File(locationString);
-			if (file.exists() && !file.isDirectory()) {
-				JarFile jar = new JarFile(file);
-				Enumeration<JarEntry> entriesS = jar.entries();
-				while (entriesS.hasMoreElements()) {
-					JarEntry entryB = entriesS.nextElement();
-					if (!entryB.isDirectory() && entryB.getName().startsWith("logisticspipes/network/guis/")) {
-						String filename = entryB.getName();
-						int classNameEnd = filename.length() - ".class".length();
-						filename = filename.substring(0, classNameEnd).replace('/', '.');
-						classInfoList.add(filename);
-					}
-				}
-			}
-
-			loadGuiProvidersFromList(classInfoList);
-
-			if (NewGuiHandler.guilist.isEmpty()) {
-				System.err.println("Fallback solution failed. Please try to move your minecraft folder to a different location.");
-				throw new RuntimeException("Cannot load GuiProvider Classes");
-			}
+			throw new RuntimeException("Cannot load GuiProvider Classes");
 		}
 	}
 
-	private static void loadGuiProvidersFromList(List<String> classes) {
-		classes.sort(Comparator.comparing(it -> it));
+	private static void loadGuiProviders(Set<Class<? extends GuiProvider>> classes) {
+		// classes.sort(Comparator.comparing(it -> it));
```

Comment:
This is needed. Best to make the Set into a List and change the comparator to Class::getName .